include/oneapi/dpl/internal/distributed_ranges_impl/detail/ranges_shim.hpp

```diff
@@ -0,0 +1,38 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#ifndef DR_USE_RANGES_V3
+
+// see: https://community.intel.com/t5/Intel-oneAPI-DPC-C-Compiler/icpx-std-views-zip-compile-error/m-p/1577247
+// needs this workaround: https://github.com/gcc-mirror/gcc/commit/be34a8b538c0f04b11a428bd1a9340eb19dec13f
+// llvm issue: https://github.com/llvm/llvm-project/issues/61763
+//#include <ranges>
+#    include "fixranges"
+
+namespace rng = ::std::ranges;
```

Comment:
I would add "std" to the alias name (e.g., `stdrng` or `stdr`) to make it clear that the alias refers to the standard functionality.

Reply:
Good point.  Should also add a `__` prefix to it as well? This is nested in the global namespace.

Reply:
> Should also add a __ prefix to it as well?

I do not think this is necessary. The use of underscores comes from oneDPL being a LLVM PSTL derivative; also we expected that at some point we might want to contribute the SYCL implementation to the upstream, so tried to keep the uglification of internal names. For the parts of oneDPL which are unlikely to get used by the GCC/LLVM std libraries, leading underscores are not required.

Reply:
I don't want this alias to be stdrng, because oneDPL code will be also included from distributed-ranges repo. Distributed-ranges repo uses ranges-v3 and some compilation define on oneDPL will be needed to allow specifying user defined namespace alias. This will be added later, after merge, when we will be removing duplicated code from DR repo. For now let's don't rename.

Reply:
In case you use ranges-v3, you do not use standard ranges, right? So ranges-v3 is sort of a substitute for standard ranges where those are absent. From that viewpoint, the proposed renaming is completely fine.

The goal is to make it clear that this namespace comes from outside, i.e. it's not `oneapi::dpl::ranges` or `oneapi::dpl::experimental::dr::ranges` etc.

Reply:
ok, applied in https://github.com/oneapi-src/oneDPL/pull/1703